packages/beacon-node/src/chain/validation/attestation.ts

```diff
@@ -510,27 +510,31 @@ async function validateAttestationNoSignatureCheck(
   }
 
   // no signature check, leave that for step1
-  const indexedAttestationContent = {
+  const indexedAttestation: IndexedAttestation = {
     attestingIndices,
     data: attData,
     signature,
   };
-  const indexedAttestation =
-    ForkSeq[fork] >= ForkSeq.electra
-      ? (indexedAttestationContent as electra.IndexedAttestation)
-      : (indexedAttestationContent as phase0.IndexedAttestation);
 
-  const attestationContent = attestationOrCache.attestation ?? {
-    aggregationBits,
-    data: attData,
-    committeeIndex,
-    signature,
-  };
-
-  const attestation =
-    ForkSeq[fork] >= ForkSeq.electra
-      ? (attestationContent as SingleAttestation<ForkPostElectra>)
-      : (attestationContent as SingleAttestation<ForkPreElectra>);
+  let attestation: SingleAttestation;
+  if (attestationOrCache.attestation) {
+    attestation = attestationOrCache.attestation;
+  } else {
+    if (!isForkPostElectra(fork)) {
```

Comment:
we could consider adding a `isPostElectra` variable for this as we call `isForkPostElectra()` a lot in the validation function

Reply:
https://github.com/ChainSafe/lodestar/pull/7265  reduced the number of `isForkPostElectra` calls quite a bit, I think this is fine